presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConnectorAPI.java

```diff
@@ -0,0 +1,127 @@
+                .httpMethod(HttpMethod.GET)
+                .accept(MediaType.TEXT_PLAIN)
+                .build();
+
+        return forwardRequest(scope, apiRequester, nodeId);
+    }
+
+    @GET
+    @Path("/{file}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response getConnectorFile(
+            @PathParam("file") String file,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConnectorAPI.class)
+                .pathMethod("getConnectorFile")
+                .httpMethod(HttpMethod.GET)
+                .resolveTemplate("file", file)
+                .accept(MediaType.TEXT_PLAIN)
+                .build();
+
+        return forwardRequest(scope, apiRequester, nodeId);
+    }
+
+    @GET
+    @Path("/{file}/{property}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response getConnectorProperty(
+            @PathParam("file") String file,
+            @PathParam("property") String property,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConnectorAPI.class)
+                .pathMethod("getConnectorProperty")
+                .httpMethod(HttpMethod.GET)
+                .resolveTemplate("file", file)
+                .resolveTemplate("property", property)
+                .accept(MediaType.TEXT_PLAIN)
+                .build();
+
+        return forwardRequest(scope, apiRequester, nodeId);
+    }
+
+    @POST
+    @Path("/{file}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response setConnectorFileByURL(
+            String url,
+            @PathParam("file") String file,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConnectorAPI.class)
+                .pathMethod("setConnectorFileByURL")
+                .httpMethod(HttpMethod.POST)
+                .resolveTemplate("file", file)
+                .accept(MediaType.TEXT_PLAIN)
+                .entity(Entity.entity(url, MediaType.TEXT_PLAIN))
+                .build();
+
+        return forwardRequest(scope, apiRequester, nodeId);
+    }
+
+    @DELETE
+    @Path("/{file}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response deleteConnectorFile(
+            @PathParam("file") String file,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConnectorAPI.class)
+                .pathMethod("deleteConnectorFile")
+                .httpMethod(HttpMethod.DELETE)
+                .resolveTemplate("file", file)
+                .accept(MediaType.TEXT_PLAIN)
+                .build();
+
```

Comment:
As a note: Don't forget to add an API for updating a particular property in the config or connector file.